packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -567,10 +567,21 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                 // And then store the tree
                 return new TreeTreeEntry(key, snapshot);
             }));
-
         return entries;
     }
 
+    public getReferencedRoutes(): string[] {
+        const referencedRoutes: string[] = [];
+        this.contexts.forEach((value: IChannelContext) => {
+            const routes = value.getReferencedRoutes();
```

Comment:
same feedback as in container - we need to use prior results (from prior snapshot) if DDS does not need to be loaded because it did not change.
We have sort-of chicken-and-egg problem. On one hand, we need to run summarizer logic as it knows if certain things actually changed. But on order to summarize, we need "is it referenced" bit (to be written into summary), which requires summarizing logic to run.

Reply:
Exactly! Like I mentioned in some of previous comments, I am trying to work out a solution here that requires minimum work and is simple from a design point of view.